### ISSUE TITLE
AddHostObjectToScript gives an AccessViolationException when called on a winrt object

### ISSUE THREAD

**Issue by krschau** [MicrosoftEdge/WebView2Feedback#2144]
**Description**
When calling AddHostObjectToScript in WinUI, you have to use the ICoreWebView2Interop interface. If you don't, a NotImplementedException is expected, but an access violation is happening instead. (See error from #2015.)

**Version**
SDK: 1.0.1020.46
Runtime: 98.0.1108.43
Framework: WinUI 2 and 3
OS: Win11

**Repro Steps**
``` cs
var core_wv2 = MyWebView2.CoreWebView2;

try
{
    core_wv2.AddHostObjectToScript("bridge", new Bridge());  // This line should fail and go to catch, instead it crashes
}
catch (NotImplementedException)
{
    // expect a NotImplementedException for now
    winrtNotImpl = true;
}

// There is an interop we can use to access the method
var interop = (ICoreWebView2Interop)(object)core_wv2;
interop.AddHostObjectToScript("bridge", new Bridge());  // This should and does work
```

[AB#38097645](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/38097645)

**Comment by champnic** [comment]
Thanks Kristen - I've added this to our backlog to get fixed :)

**Comment by champnic** [comment]
Fixed in WinAppSDK 1.0.3 and 1.1-preview2. Thanks!
